model/src/main/java/com/cloudera/nav/sdk/model/relations/Relation.java

```diff
@@ -178,11 +195,11 @@ public T ep2(Collection<Entity> ep2Entities) {
   private String identity;
   @MProperty(required=true)
   private RelationType type;
-  @MProperty(required=true, attribute="endpoint1Ids")
```

Comment:
Have added a **validateRelation** method in RelationValidator which ensures that a Relation either the id or the attributes needed to generate the id.